addons/sale_project/models/sale_order.py

```diff
@@ -36,10 +36,12 @@ def _compute_visible_project(self):
     @api.depends('order_line.product_id', 'order_line.project_id')
     def _compute_project_ids(self):
         is_project_manager = self.user_has_groups('project.group_project_manager')
+        Project = self.env['project.project']
         for order in self:
             projects = order.order_line.mapped('product_id.project_id')
             projects |= order.order_line.mapped('project_id')
             projects |= order.project_id
+            projects |= Project.search([('sale_line_id', 'in', order.order_line.ids)])
```

Comment:
And finally we can get the projects linked to the SO like this:
```suggestion
            projects |= projects_per_so[order.id or order._origin.id]
```